trace/trace_test.go

```diff
@@ -63,6 +63,50 @@ func TestStep(t *testing.T) {
 	}
 }
 
+func TestNestedTrace(t *testing.T) {
+	tests := []struct {
+		name          string
+		inputString   string
+		expectedTrace *Trace
+	}{
+		{
+			name:        "Empty string",
+			inputString: "",
+			expectedTrace: &Trace{
+				nestedTrace: []*Trace{
+					{startTime: time.Now(), name: ""},
+				},
+			},
+		},
+		{
+			name:        "Non-empty string",
+			inputString: "Inner trace",
+			expectedTrace: &Trace{
+				nestedTrace: []*Trace{
+					{
+						startTime: time.Now(),
+						name:      "Inner trace",
+						nestedTrace: []*Trace{
+							{name: "Inner trace"},
+						},
+					},
+				},
+			},
+		},
```

Comment:
I case the verifies the sorting of `traceSteps` and `Trace` works as expected seems valuable.